src/services/AdvancedTradingView/store/advancedTradingView.reducer.ts

```diff
@@ -58,6 +64,58 @@ const advancedTradingViewSlice = createSlice({
         },
       }
     },
+
+    setBurnsAndMintsDataForAllPairs: (
+      state: InitialState,
+      action: PayloadAction<Array<SetBurnsAndMintsActionPayload>>
+    ) => {
+      const updatedAdapters: AdapterType = {
+        ...state.adapters,
+      }
+
+      action.payload.forEach(adapter => {
+        const { key, pairId, data, hasMore } = adapter
+
+        const previousPairData = state.adapters[key][pairId]?.['burnsAndMints']?.data ?? []
+
+        data.forEach(element => !previousPairData.some(el => el.id === element.id) && previousPairData.push(element))
+
```

Comment:
Nice! I would name `previousPairData` `nextPairData` to avoid confusion because in next line you add data to it :D and push it back to the slice

Reply:
I copied the name from function above but I agree. 👍🏼